### PR TITLE
Feature: Feature: Enhance Performance, Logging & Stability: Precompile Regex, Command Matching, Golang 1.24, History Cleanup & memLimitMiB Flag.

### PR DESCRIPTION
**Pull request by bryannolen**
* pre-compile regex at config load, allowing errors to be found faster.
    This also speeds up regexp matching, especially for complex values.
* Update CI and CodeQL to use Go 1.24.1 (required setting the "quality gate" to 55% due to the coverage calculation method being fixed in Go 1.21)
* Add a cleanup/ageout routine to HistoryStore.

See https://github.com/mariocandela/beelzebub/discussions/186 for full change list!

 All Submissions:

* [ ✅ ] Have you followed the guidelines in our Contributing document?
* [ ✅ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ✅ ] Does your submission pass tests?
2. [ ✅ ] Have you lint your code locally before submission?

### Changes to Core Features:

* [ ✅ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ✅ ] Have you written new tests for your core changes, as applicable?
* [ ✅ ] Have you successfully run tests with your changes locally?

## Summary by CodeRabbit

- **New Features**
  - Introduced a command-line option to set a process memory limit (default 100 MiB) for better resource management.
  - Implemented automatic cleanup of outdated session histories to ensure a more responsive experience.
  - Enhanced configuration processing with improved command pattern validation for more reliable service setups.
  - Streamlined network interactions with refined tracing and command handling in both HTTP and SSH protocols.

### CODE REVIEW

**Comment by bryannolen** [comment]
Example of named command:
```
  - regex: "^whoami$"
    handler: "root"
    name: "IAmRoot"
```

Will add the field `Handler: "IAmRoot"` to the logged event.

**Comment by mariocandela** [comment]
> Example of named command:
> 
> ```
>   - regex: "^whoami$"
>     handler: "root"
>     name: "IAmRoot"
> ```
> 
> Will add the field `Handler: "IAmRoot"` to the logged event.

This feature is interesting 💯  for future ideas, I'd ask you to first submit them in the issues(ideas) section so that we can discuss them all together. 
Thank you very much for your contribution.

**Comment by bryannolen** [comment]
I have updated the feature notes accordingly.

In future I'll propose the idea in the discussion section before just dropping a PR :)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: CI
 
 on:
   push:
-    branches: [ main ]
+    branches: [ "main", "future" ]
```
Comment (mariocandela) [inline .github/workflows/ci.yml:5]:
```suggestion
    branches: [ "main" ]
```

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: "CodeQL"
 
 on:
   push:
-    branches: [ "main" ]
+    branches: [ "main", "future" ]
```
Comment (mariocandela) [inline .github/workflows/codeql.yml:5]:
```suggestion
    branches: [ "main" ]
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -35,7 +35,7 @@ jobs:
 
     - name: Quality Gate - Test coverage shall be above threshold
       env:
-        TESTCOVERAGE_THRESHOLD: 65
+        TESTCOVERAGE_THRESHOLD: 55
```
Comment (mariocandela) [inline .github/workflows/ci.yml:38]:
```suggestion
        TESTCOVERAGE_THRESHOLD: 80
```

File: go.mod [hunk 4]
```diff
@@ -1,6 +1,8 @@
 module github.com/mariocandela/beelzebub/v3
 
-go 1.20
+go 1.23.0
```
Comment (mariocandela) [inline go.mod:3]: Why into continuous integration you use 1.24.1 ? But into go.mod 1.23.0 ? It's a typo ? 😄

**Review by mariocandela — CHANGES_REQUESTED** [review]
I just need to test a few edge use cases, and then I'll merge it! Thank you so much for the contribution! 🚀 

ps: check your email I replied to you 😄

**Comment by bryannolen** [comment]
Changes made, the 1.20 test will fail as `go.mod` added the `toolchain` directive  in 1.21.

Also I don't see a reply from you to that email?

**Comment by mariocandela** [comment]
@coderabbitai full review

File: historystore/history_store.go [hunk 5]
```diff
@@ -41,7 +53,30 @@ func (hs *HistoryStore) Append(key string, message ...plugins.Message) {
...
+			hs.Lock()
+			for k, v := range hs.sessions {
+				if time.Since(v.LastSeen) > MaxHistoryAge {
+					delete(hs.sessions, k)
+				}
+			}
+			hs.Unlock()
+		}
+	}()
 }
```
[the remark this answers is not in the document]
Comment (bryannolen) [reply historystore/history_store.go:82]: This is its exact purpose - to run forever in a long running process.

**Review by mariocandela — APPROVED** [review]
LGTM, Great job, much appreciated 💯
